shotover-proxy/src/frame/cassandra.rs

```diff
@@ -462,38 +515,512 @@ impl CassandraOperation {
+                            return true;
+                        }
+                    }
+                }
+            }
+            CassandraStatement::Select(select) => {
+                return CQLStatement::has_params_in_relation_elements(&select.where_clause);
+            }
+            CassandraStatement::Update(update) => {
+                for assignment_element in &update.assignments {
+                    if let Operand::Param(_) = &assignment_element.value {
+                        return true;
+                    }
+                    if let Some(assignment_operator) = &assignment_element.operator {
+                        match assignment_operator {
+                            AssignmentOperator::Plus(operand) => {
+                                if let Operand::Param(_) = operand {
+                                    return true;
+                                }
+                            }
+                            AssignmentOperator::Minus(operand) => {
+                                if let Operand::Param(_) = operand {
+                                    return true;
+                                }
+                            }
+                        }
+                    }
+                }
+                if CQLStatement::has_params_in_relation_elements(&update.where_clause) {
+                    return true;
+                }
+                if CQLStatement::has_params_in_relation_elements(&update.if_clause) {
+                    return true;
+                }
+            }
+            _ => {}
+        }
+        false
+    }
+}
+
+impl Display for CQLStatement {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        self.statement.fmt(f)
+    }
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct CQL {
+    pub statements: Vec<Box<CQLStatement>>,
+    pub(crate) has_error: bool,
+}
+
+impl CQL {
+    /// the number of statements in the CQL
+    pub fn get_statement_count(&self) -> usize {
+        self.statements.len()
+    }
+
+    fn from_value_and_col_spec(value: &Value, col_spec: &ColSpec) -> Operand {
+        match value {
+            Value::Some(vec) => {
+                let cbytes = CBytes::new(vec.clone());
+                let message_value =
+                    MessageValue::build_value_from_cstar_col_type(col_spec, &cbytes);
+                let pmsg_value = &message_value;
+                pmsg_value.into()
+            }
+            Value::Null => Operand::Null,
+            Value::NotSet => Operand::Null,
+        }
+    }
+    fn set_param_value_by_name(
+        name: &str,
+        query_params: &QueryParams,
+        param_types: &[ColSpec],
+    ) -> Operand {
+        if let Some(QueryValues::NamedValues(value_map)) = &query_params.values {
+            if let Some(value) = value_map.get(name) {
+                if let Some(idx) = value_map
+                    .iter()
+                    .enumerate()
+                    .filter_map(
```

Comment:
This logic is a little confusing to me. It just checks the first item of the iterator?

Reply:
So the `filter_map` operation is only performed once?